packages/deployments/contracts/contracts_forge/PromiseMessage.t.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+import "./ForgeHelper.sol";
+
+import "../contracts/interfaces/IConnext.sol";
+import "../contracts/nomad-xapps/contracts/promise-router/PromiseMessage.sol";
+import {Home} from "../contracts/nomad-core/contracts/Home.sol";
+
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+// running tests (with logging on failure):
+// yarn workspace @connext/nxtp-contracts test:forge -vvv
+// run a single test:
+// yarn workspace @connext/nxtp-contracts test:forge -m testAddRouterAlreadyApproved -vvv
+
+// other forge commands: yarn workspace @connext/nxtp-contracts forge <CMD>
+// see docs here: https://onbjerg.github.io/foundry-book/index.html
+
+contract PromiseMessageTest is ForgeHelper {
+  // ============ Libraries ============
+  using TypedMemView for bytes;
+  using TypedMemView for bytes29;
+  using PromiseMessage for bytes29;
+
+  // ============ Test set up ============
+
+  function setUp() public {}
+
+  // ============ format/parse ============
+  // Should work
+  function test_PromiseCallbackMessage_should_format_and_parse_a_massage_properly(
```

Comment:
just reformatted the names to be follow the structure: `test_${ContractName}__${functionName}_${testName}`

which means this test name should be:

`test_PromiseMessage__formatMessage_shouldWork`

or something (test name is a little tricky here, but the main thing is camel over snake casing)

Reply:
all renamed!
Good structure!